segfast/utils.py

```diff
@@ -35,6 +37,112 @@ def update(self, n=1):
+
+        dtype = dtype or self.STANDARD_BYTE_TO_LEN[self.start_byte]
+        if isinstance(dtype, int):
+            dtype = 'i' + str(dtype)
+
+        self.dtype = np.dtype(dtype)
+        self.default_byteorder = byteorder
+        self.has_explicit_byteorder = isinstance(dtype, str) and dtype[0] in {'>', '<'}
+        if not self.has_explicit_byteorder and byteorder is not None:
+            self.dtype = self.dtype.newbyteorder(byteorder)
+
+        if self.start_byte + self.byte_len > self.TRACE_HEADER_SIZE + 1:
+            raise ValueError(f'{self.name} header position is out of bounds')
+
+    @property
+    def byte_len(self):
+        """ The number of bytes for a header. """
+        return self.dtype.itemsize
+
+    @property
+    def is_standard(self):
+        """ Whether the header matches the specification. """
+        return self.name in self.STANDARD_BYTE_TO_HEADER and self.has_standard_location
+
+    @property
+    def has_standard_location(self):
+        """ Whether the header matches the specification, except maybe the name. """
+        return self.start_byte in self.STANDARD_BYTE_TO_HEADER and \
+               self.byte_len == self.STANDARD_BYTE_TO_LEN[self.start_byte] and \
+               np.issubdtype(self.dtype, np.integer)
+
+    @property
+    def standard_name(self):
+        """ The name from specification for header (if 'has_standard_location' is True). """
+        if not self.has_standard_location:
+            raise ValueError("The header has non-standard start byte or dtype")
+        return self.STANDARD_BYTE_TO_HEADER[self.start_byte]
+
+    @property
+    def has_default_byteorder(self):
+        """ Whether default byteorder is defined. """
+        return self.default_byteorder is not None
+
+    @property
+    def has_byteorder(self):
+        """ Whether byteorder is defined. """
+        return self.has_explicit_byteorder or self.has_default_byteorder
+
+    @property
+    def byteorder(self):
+        """ Header byteorder (if defined). """
+        if not self.has_byteorder:
+            return None
+        return self.dtype.str[0]
+
+    @property
+    def _spec_params(self):
+        dtype_str = self.dtype.str
+        if not self.has_byteorder:
+            dtype_str = dtype_str[1:]
+        return self.name, self.start_byte, dtype_str
+
+    def __eq__(self, other):
+        return self._spec_params == other._spec_params
+
+    def __hash__(self):
+        return hash(self._spec_params)
+
+    def set_default_byteorder(self, byteorder):
+        """ Set byteorder to use as default (if not specified by dtype). """
+        dtype = self.dtype.str
+        if not self.has_explicit_byteorder:
+            dtype = dtype[1:]
+        return type(self)(name=self.name, start_byte=self.start_byte, dtype=dtype, byteorder=byteorder)
+
+    def __repr__(self):
+        return f"{type(self).__name__}(name='{self.name}', start_byte={self.start_byte}, dtype='{self.dtype.str}')"
```

Comment:
Please use dtype string like that in `_spec_params` instead of `self.dtype.str`: endianness should be shown only if it was explicitly given.